src/dictionaries.cpp

```diff
@@ -41,28 +43,41 @@ Dictionaries::Dictionaries(const std::string& filePath)
     if (!stream)
         throw std::runtime_error("Cannot open dictionary file " + filePath);
 
+    using tokenizer =  boost::tokenizer<boost::char_separator<char>>;
+    boost::char_separator<char> sep("\t");
```

Comment:
I read the doc carefully and also I did some experiments. First, from the doc, there are two constructors for `boost_char_separator`. One with arguments and one default (no arguments). The default one creates a separator which use `std::isspace` for droppped delimeters and `std::ispunct` for kept delimeters. It means that this constructor will tokenize by space-like characters (whitespace, tabulation, etc.) and punctuation (coma, semicolon, etc.). It will drop the whitespaces but keep the punctuation. The following code:
```c++
int main()
{
  std::string str = "Foo , bar,    baz,     huzzy";
  using tokenizer = boost::tokenizer<boost::char_separator<char>>;
  boost::char_separator<char> sep;
  tokenizer tokens(str, sep);
  for (const auto& tok : tokens)
      std::cout << "<" << tok << "> ";
  std::cout << "\n";
  return EXIT_SUCCESS;
}
```
will result in the following output:
```
<Foo> <,> <bar> <,> <baz> <,> <huzzy>
```
(notice that the words do not contain spaces, but punctuation is kept).

The other constructor have 3 arguments: dropped delimiters, kept delimiters and empty token policy. First argument contains a list of delimiters that will be dropped (like whitespaces in the previous example). Second argument contains a list of delimiters that will be kept (like punktuation in the example above). Third argument lets you decide whether to keep empty tokens (e.g., the string ",," has 3 empty tokens if we use ',' as a delimiter). By default, the second argument is an empty string (no kept tokens) and the third is configured to drop empty tokens. So you only need to specify the first argument. Here is an example:
```c++
int main()
{
  std::string str = "Foo , bar,    baz,     huzzy";
  using tokenizer = boost::tokenizer<boost::char_separator<char>>;
  boost::char_separator<char> sep(",");
  tokenizer tokens(str, sep);
  for (const auto& tok : tokens)
      std::cout << "<" << tok << "> ";
  std::cout << "\n";
  return EXIT_SUCCESS;
}
```
The output is:
```
<Foo > < bar> <    baz> <     huzzy>
```
See how the words were separated by comas, but the whitespaces are kept in place.

To drop whitespaces you need to specify them in the delimiter list:
```c++
int main()
{
  std::string str = "Foo , bar,    baz,     huzzy";
  using tokenizer = boost::tokenizer<boost::char_separator<char>>;
  boost::char_separator<char> sep(", ");
  tokenizer tokens(str, sep);
  for (const auto& tok : tokens)
      std::cout << "<" << tok << "> ";
  std::cout << "\n";
  return EXIT_SUCCESS;
}
```
The result is a nice list of words without any whitespaces and punctuation:
```
<Foo> <bar> <baz> <huzzy>
```

So in your case if you want to split line by whitespaces you may either use the default constructor (like in the first example) or specify all space-like symbols, not only the tabulation.